docs/src/getting_started.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Getting Started Guide
+slug: /getting_started
+---
+
+# Getting Started with Charmil
+
+This guide assumes that you already have a working Go environment, if not then please refer to
+[this page](https://golang.org/doc/install) first.
+
+## Charmil CLI
+
+Charmil offers its own command-line interface (CLI) that allows developers to construct their new Charmil project with additional support for other templates, allowing them to focus on more essential aspects of their project.
+&nbsp;
+
+- ### Installation:
+
+  In order to use the Charmil CLI, run the following command:
+
+  ```bash
+  $ go get github.com/aerogear/charmil/cli/cmd/charmil
+  ```
+
+  This will create the Charmil CLI executable under your `$GOPATH/bin` directory.
+
+- ### Usage:
+  For usage instructions on Charmil CLI, refer to [this link](./charmil_cli.md).
+
+## Charmil Starter
+
+The Charmil Starter Template allows users to quickly create plugins and host commands that make use of the features offered by Charmil. This template includes all of the tools and solutions needed to create a command-line utility.
+&nbsp;
+
+- ### Usage:
+
+  The Charmil Starter Template can be used by any of the following ways:
+
+  - By using the `charmil init` command offered by the Charmil CLI
+    ```bash
+    $ charmil init
+    ```
+  - By cloning the [Charmil Starter Github repository](https://github.com/aerogear/charmil-starter)
+    ```bash
+    $ git clone https://github.com/aerogear/charmil-starter.git
+    ```
+
+## Charmil SDK
```

Comment:
Add those as separate chapters. we need guide for starter

Reply:
I've updated the structure now. Is it fine?

> we need guide for starter extented..How to create command. Adding elements to config, crud etc.

Yes, I'll add all of this to the starter doc once we're done with the cleanup of starter and core